### DISCUSSION TITLE
Team B - Describe components

### DISCUSSION THREAD

**Discussion by mindgitrwx** [lablup/contribution-academy-2021#80] [Show and tell]
Describe the components of Backend.ai in one sentence.

**Comment by TeusEE** [comment]
backend.ai-storage-proxy (Storage Proxy) : common component의 Asyncetcd를 사용해서 backend.ai-manager와 client의 요청을 받고, 해당 요청에 대한 fileSystem 구축(이부분은 약간 이해를 못했습니다;;)

backend.ai-webserver (Web Server) : aiohttp를 사용해서 사용지 인증 처리, command line의 log 발생, Client component의 aiohttp requests 요청에 대한 RestAPI 구축.

**Comment by min1826** [comment]
* **webUI**: 사용자가 GUI 인터페이스를 쉽게 이용할 수 있도록 제공하는 데스크톱 앱과 웹서비스
* **client-py**: Python을 위한 backend.ai API를 제공하고, backend.ai CLI 역할을 함

**Comment by AlchemistX** [comment]
- Kernels : Agent를 통해 실행하는 Docker Image를 생성하는 설정파일들의 모임으로 각종 개발 환경, 언어, 프레임워크들의 런타임 구성들을 정의한다. 
- Common : 모든 컴포넌트에서 공통으로 사용하는 기능들의 집합.

**Comment by mindgitrwx** [comment]
**Agent**      : 커널 컨테이너의 라이프사이클을 관리, worker의 상태를 체크. 
**Manager** :  Agent와 계산 커널의 추상, User 와 Admin API (REST/GraphQL)  aiohttp 기반 게이트웨이

사실 Manager의 경우 README에 적혀있는 설명만으로는 명확하게 이해가 되지 않았고,  인증이 안되는 시행착오 중에 Python에러에서 stacktrace 를 따라가며 Manager에서 Graphql을 사용하는 부분을 발견, manager에서 Graphql이 어떻게 적용되는가 확인해 봄.
